prow-configurations/install-prow.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -o errexit
+
+kubectl create clusterrolebinding cluster-admin-binding \
+  --clusterrole cluster-admin --user $(gcloud config get-value account)
+
+# Deploy NGINX Ingress Controller
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.20.0/deploy/mandatory.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.20.0/deploy/provider/cloud-generic.yaml
+
+# Follow the installation steps in https://github.com/kubernetes/test-infra/blob/master/prow/getting_started.md#how-to-turn-up-a-new-cluster
+
+## Create an HMAC token
+hmac_token="$(openssl rand -hex 20)"
+echo "hmac-token: $hmac_token"
+
+if [ "$OAUTH" == "" ]; then
+    echo -n "Enter OAuth2 token that has read and write access to the bot account, followed by [ENTER]: "
+    read oauth_token
+else
+    oauth_token="$OAUTH"
+fi
+
+kubectl create secret generic hmac-token --from-literal=hmac=$hmac_token
+kubectl create secret generic oauth-token --from-literal=oauth=$oauth_token
+
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/test-infra/a202e595a33ac92ab503f913f2d710efabd3de21/prow/cluster/starter.yaml
+
+# Add annotations to Prow Ingress 
+kubectl annotate ingress ing kubernetes.io/ingress.class=nginx nginx.ingress.kubernetes.io/ssl-redirect=false
```

Comment:
To discuss: can we make this scripts configurable, and configure it by namespace where PROW will be installed?
Would it be possible to install many prows on one cluster? Maybe it will be an option, instead of provisioning whole cluster for testing our prow configuration, to provision prow on the same cluster but in a different namespace

Reply:
It looks problematic to install Prow on a namespace because of this issue https://github.com/kubernetes/test-infra/issues/9394.